src/shared/molecules/MyDataHeader/MyDataHeaderFilters/TypeSelector.tsx

```diff
@@ -0,0 +1,211 @@
+import React, { useCallback, useRef, useState } from 'react';
+import { NexusClient } from '@bbp/nexus-sdk';
+import { useNexusContext } from '@bbp/react-nexus';
+import { Checkbox, Col, Dropdown, Input, Row, Select } from 'antd';
+import { concat, isString, startCase } from 'lodash';
+import { useQuery } from 'react-query';
+import {
+  THeaderProps,
+  TType,
+  TTypeAggregationsResult,
+  TTypesAggregatedBucket,
+} from '../../../canvas/MyData/types';
+import useMeasure from '../../../hooks/useMeasure';
+import isValidUrl from '../../../../utils/validUrl';
+import { prettifyNumber } from '../../../../utils/formatNumber';
+
+const getTypesByAggregation = async ({
+  nexus,
+  orgLabel,
+  projectLabel,
+}: {
+  nexus: NexusClient;
+  orgLabel?: string;
+  projectLabel?: string;
+}) => {
+  return await nexus.Resource.list(orgLabel, projectLabel, {
+    aggregations: true,
+  });
+};
+
+const useTypesAggregation = ({
+  nexus,
+  selectCallback,
+}: {
+  nexus: NexusClient;
+  selectCallback: (data: any) => TType[];
+}) => {
+  return useQuery({
+    refetchOnWindowFocus: false,
+    queryKey: ['types-aggregation-results'],
+    queryFn: () => getTypesByAggregation({ nexus }),
+    select: selectCallback,
+  });
+};
+const typesOptionsBuilder = (typeBucket: TTypesAggregatedBucket): TType => {
+  const typeKey = typeBucket.key;
+  const typeLabel =
+    isString(typeKey) && isValidUrl(typeKey)
+      ? typeKey.split('/').pop()
+      : typeKey;
+
+  return {
+    key: typeKey,
+    value: typeKey,
+    label: startCase(typeLabel),
+    docCount: typeBucket.doc_count,
+  };
+};
+
+const TypeRowRenderer = ({
+  type,
+  checked,
+  onCheck,
+}: {
+  checked: boolean;
+  type: TType;
+  onCheck(e: React.MouseEvent<HTMLElement, MouseEvent>, type: TType): void;
+}) => {
+  return (
+    <Row justify="space-between" align="top" key={type.key}>
+      <Col span={20}>
+        <span title={`${type.value}, (${type.docCount})`}>{type.label}</span>
+      </Col>
+      <Col
+        span={4}
+        style={{
+          display: 'flex',
+          justifyContent: 'flex-end',
+          alignItems: 'center',
+        }}
+      >
+        <Checkbox onClick={e => onCheck(e, type)} checked={checked} />
+      </Col>
+    </Row>
+  );
+};
+const TypeSelector = ({
+  types,
+  setFilterOptions,
+}: Pick<THeaderProps, 'types' | 'setFilterOptions'>) => {
+  const nexus = useNexusContext();
+  const originTypes = useRef<TType[]>([]);
+  const [typeSearchValue, updateSearchType] = useState('');
+  const [typesOptionsArray, setTypesOptionsArray] = useState<TType[]>([]);
+
+  const selectCallback = useCallback((data: any) => {
+    const options = (
+      ((data as unknown) as TTypeAggregationsResult).aggregations.types
```

Comment:
imho we should do this translation of types (`data as unknown as TTypeAggregationsResult` in the `useAggregations` hook. The user of the this hook should be able to safely guess the type as `TTypeAggregationsResult` from the value returned by the hook:

```javascript
const useTypesAggregation = ({
  nexus,
  selectCallback,
}: SomeType) => {
  return useQuery({
    ...
    select: (data) => selectCallback(data as unknown as TTypesAggregatedBucket),
  });
};

// user of the hook
  const selectCallback = useCallback((data: TTypeAggregationsResult) => {
    const options = (
    data.aggregations.types
        ?.buckets ?? ([] as TTypesAggregatedBucket[])
    )
    ...
  }, []);
    const { data: typeOptions } = useTypesAggregation({
    nexus,
    selectCallback,
  });
```